core/chains/evm/gas/models.go

```diff
@@ -54,7 +55,7 @@ type feeEstimatorClient interface {
 }
 
 // NewEstimator returns the estimator for a given config
-func NewEstimator(lggr logger.Logger, ethClient feeEstimatorClient, chaintype chaintype.ChainType, geCfg evmconfig.GasEstimator) (EvmFeeEstimator, error) {
+func NewEstimator(lggr logger.Logger, ethClient feeEstimatorClient, chaintype chaintype.ChainType, geCfg evmconfig.GasEstimator, getChainClientByID func(id string) (evmclient.Client, error)) (EvmFeeEstimator, error) {
```

Comment:
Why pass a function instead of the actual map?

Reply:
I feel that returning the `evmclient.Client` interface creates a huge dependency on the estimator's side. Don't we already know the method we will use from the Client? Can't we declare a much smaller interface in its place?